bpf/tests/wireguard_native.c

```diff
@@ -0,0 +1,242 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Cilium */
+
+#include <bpf/ctx/skb.h>
+#include "common.h"
+#include "pktgen.h"
+
+#define ENABLE_IPV4			1
+#define ENABLE_WIREGUARD		1
+#define ENABLE_IDENTITY_MARK		1
+
+#define SRC_POD_V4			v4_pod_one
+#define DST_POD_V4			v4_pod_one_on_node_two
+#define DST_POD_CIDR_V4			v4_pod_cidr_on_node_two
+
+#define SRC_POD_SEC_IDENTITY		(CIDR_IDENTITY_RANGE_START - 2)
+#define DST_POD_SEC_IDENTITY		(CIDR_IDENTITY_RANGE_START - 3)
+
+#define SRC_NODE_V4			v4_node_one
+#define DST_NODE_V4			v4_node_two
+
+#define ENCRYPT_KEY			0xFF
+
+#include "bpf_host.c"
+
+#include "lib/ipcache.h"
+
+#define TO_NETDEV 0
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PROG_ARRAY);
+	__uint(key_size, sizeof(__u32));
+	__uint(max_entries, 1);
+	__array(values, int());
+} entry_call_map __section(".maps") = {
+	.values = {
+		[TO_NETDEV] = &cil_to_netdev,
+	},
+};
+
+static __always_inline
+int v4_build_packet(struct __ctx_buff *ctx)
+{
+	struct pktgen builder;
+	struct iphdr *l3;
+
+	pktgen__init(&builder, ctx);
+
+	l3 = pktgen__push_ipv4_packet(&builder, (__u8 *)mac_one, (__u8 *)mac_two,
```

Comment:
Just fyi, here it doesn't seem needed (don't need to manipulate the packet too much), but in  https://github.com/cilium/cilium/pull/40294 we added a new way to define packets.